xds/internal/xdsclient/xds.go

```diff
@@ -561,6 +562,33 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger,
 	return routesRet, nil
 }
 
+func hashPoliciesProtoToSlice(policies []*v3routepb.RouteAction_HashPolicy) []*HashPolicy {
+	var hashPoliciesRet []*HashPolicy
+	for _, p := range policies {
+		var policy HashPolicy
+		policy.Terminal = p.Terminal
+		switch p.GetPolicySpecifier().(type) {
+		case *v3routepb.RouteAction_HashPolicy_Header_:
+			policy.HashPolicyType = HashPolicyTypeHeader
+			policy.HeaderName = p.GetHeader().HeaderName
+			policy.Regex = p.GetHeader().GetRegexRewrite().GetPattern().Regex
```

Comment:
Make these two type `*regexp.Regexp`, parse the regex, and return error if it returns errors.

Like https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xds.go#L461

Reply:
Ah, I treated these errors as no-ops in config selector. Your idea is much better, fail at parsing.